lib/friendly_shipping/services/usps_ship.rb

```diff
@@ -141,7 +142,9 @@ def rate_estimates(shipment, options:, debug: false)
                 shipping_method: rates.first.shipping_method,
                 data: rates.first.data
               )
-            ]
+            ],
+            original_request: api_results.first.original_request,
+            original_response: api_results.first.original_response
```

Comment:
Don't you need to keep the corresponding request and response with each package rate result?

Reply:
There really isn't a good way to do that, unfortunately. Each `Rate` does not hold a request or response. Only the `ApiResult` holds that, and we're combining (summing) all the rates into a single `ApiResult` to return from this method due to the way the USPS API works. So we either have to pick one request/response to return or we need to return all the requests/responses as arrays which would change the established interface for `ApiResult` objects. Given that we typically only use the request/response for debugging purposes, returning the first request/response for a successful request seems adequate to me. WDYT?